scripts/firedrake-zenodo

```diff
@@ -157,6 +160,16 @@ else:
                         format='%(message)s')
 log = logging.getLogger()
 
+# TODO: Backward compatibility. Remove this block eventually.
+src_petsc4py_exists = os.path.exists(os.path.join(os.environ["VIRTUAL_ENV"], "src/petsc4py"))
+if not args.input_file and src_petsc4py_exists:
+    if args.ignore_src_petsc4py:
+        descriptions.pop("petsc4py")
+        projects.pop("petsc4py")
+        components.remove("petsc4py")
+    else:
+        log.warning("Found src/petsc4py, but the latest version of Firedrake builds petsc4py from petsc source tree instead of from src/petsc4py. If you built your petsc4py from petsc source tree and wish to ignore src/petsc4py, use --ignore-src-petsc4py option.")
+
```

Comment:
Also fix  this for not having `src_petsc4py_exists`.

Reply:
Fixed.